router/proxy/metrics.go

```diff
@@ -0,0 +1,65 @@
+package proxy
+
+import (
+	"strconv"
+	"time"
+
+	router "github.com/flynn/flynn/router/types"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	prometheus.MustRegister(backendHTTPStatusMetric)
+	prometheus.MustRegister(backendHTTPConnErrorMetric)
+	prometheus.MustRegister(backendHTTPLatencyMetric)
+}
+
+var backendHTTPConnErrorMetric = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "router_backend_http_conn_errors_total",
+		Help: "Router backend HTTP connection errors",
+	},
+	[]string{"app", "service", "job_id"},
+)
+
+func trackBackendHTTPConnError(backend *router.Backend) {
+	backendHTTPConnErrorMetric.With(prometheus.Labels{
+		"app":     backend.App,
+		"service": backend.Service,
+		"job_id":  backend.JobID,
+	}).Inc()
+}
+
+var backendHTTPStatusMetric = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "router_backend_http_status_total",
+		Help: "Router backend HTTP status codes",
+	},
+	[]string{"app", "service", "job_id", "status"},
+)
+
+func trackBackendHTTPStatus(backend *router.Backend, status int) {
+	backendHTTPStatusMetric.With(prometheus.Labels{
+		"app":     backend.App,
+		"service": backend.Service,
+		"job_id":  backend.JobID,
```

Comment:
I think these should be `backend_app`, `backend_service`, and `backend_job_id`, and then we should also have `job_id` and maybe `host` for the router instance?